src/mtcp/mtcp_restart.c

```diff
@@ -883,8 +883,8 @@ read_one_memory_area(int fd, VA endOfStack)
    * prepareMtcpHeader() in threadlist.cpp and ProcessInfo::growStack()
    * in processinfo.cpp.
    */
-  if ((area.name[0] && area.name[0] != '/' && mtcp_strstr(area.name, "stack"))
-      || (area.endAddr == endOfStack)) {
+  if (area.name[0] && (mtcp_strstr(area.name, "[stack]") ||
```

Comment:
1. Because this is an 'and' inside an 'or', some compilers will issue warnings about including `(...)` around the 'and' clause for clarity.  So, let's keep the extra parentheses.
2.  I do like checking for `[stack]` with the `[...]`.  Thanks.
3. NITPICK:  Should we keep the clause `area.name[0] != '/'`?  In principle, some target application could use a weird filename like `myprog\[drat1\].c`.  This could happen, for example, after importing a program from some future operating system that decides on that convention for file names.

Reply:
I think the MANA code was slightly outdated, and so I picked the wrong version. I have removed this commit because I don't think we need to change anything here.